### PR TITLE
New Port: GNU FreeDink - Dink Smallwood

### PR DESCRIPTION
**Pull request by Cebion**
Title: FreeDink
URL: https://www.gnu.org/software/freedink/

DMOD Support
For DMODS download the mod from https://www.dinknetwork.com/files/category_dmod/

Extract the .dmod file and copy the extracted folder into the freedink folder

Controls
| Button | Action |
|--|--| 
|DPAD| Walk / Push|
|B| Attack / Use Selected Item|
|A| Talk / Examine / Manipulate|
|X| Use equipped magic|
|Y| Select Highlighted Item|
|R1| Map|
|Start| Inventory / Equip Screen| 
|Select| Escape| 

CFW Tests:
[X] AmberELEC
[X] ArkOS
[X] ROCKNIX
[X] MuOS
[] Knulli (Optional)
[] Crossmix (Optional)

Architectures: 
[x] Aarch64
[X] Panfrost (Mainline)

Resolutions: 
[] 480x320 (Optional)
[x] 640x480
[X] 720x720 (RGB30)
[x] Higher resolutions (e.g., 1280x720)

### CODE REVIEW

**Review by JanTrueno — APPROVED** [review]
👍

File: ports/freedink/FreeDink.sh [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+> "$GAMEDIR/log.txt" && exec > >(tee "$GAMEDIR/log.txt") 2>&1
+
+mkdir -p "$GAMEDIR/conf"
+
+export LD_LIBRARY_PATH="$GAMEDIR/libs.${DEVICE_ARCH}:$LD_LIBRARY_PATH"
+export SDL_GAMECONTROLLERCONFIG="$sdl_controllerconfig"
+export XDG_DATA_HOME="$CONFDIR"
+export SDL_SOUNDFONTS="$GAMEDIR/gs.sf2"
+
+# $ESUDO rm -rf ~/.boswars
```
Comment (cdeletre) [inline ports/freedink/FreeDink.sh:34]: It can be removed I guess ?

File: ports/freedink/FreeDink.sh [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+source $controlfolder/control.txt
+source $controlfolder/device_info.txt
+
+[ -f "${controlfolder}/mod_${CFW_NAME}.txt" ] && source "${controlfolder}/mod_${CFW_NAME}.txt"
+
+get_controls
+
+GAMEDIR=/$directory/ports/freedink
+CONFDIR="$GAMEDIR/conf/"
+BINARY=minife
```
Comment (cdeletre) [inline ports/freedink/FreeDink.sh:23]: Shouldn't we use "minife.${DEVICE_ARCH}" ? or it is no more a requirement ?

File: ports/freedink/freedink/dink/Story/1GOLD.c
Comment (cdeletre) [inline ports/freedink/freedink/dink/Story/1GOLD.c:1]: I don't know how the game works but do we need all of these C files ?

File: ports/freedink/FreeDink.sh (hunk 2 above)
Comment (Cebion) [reply ports/freedink/FreeDink.sh:23]: Requirement if a port is multiarch. I do it if it has potential for other arch.  This one I would not use for x86 for example.

File: ports/freedink/freedink/dink/Story/1GOLD.c
Comment (Cebion) [reply ports/freedink/freedink/dink/Story/1GOLD.c:1]: Looks like they're script files and not normal .c files

**Pull request merged**